File: 2. 两数相加/001.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */

public class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }
}
    class Solution {
        public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
            boolean overflow = false;
            int value;
            ListNode head=null;
            ListNode tail=null;
            while(l1!=null || l2!=null || overflow){
                value = (l1==null?0:l1.val) + (l2==null?0:l2.val) + (overflow?1:0);
                if(head==null){
                    head = new ListNode(value%10);
                    tail = head;
                }
                else{
                    tail.next = new ListNode(value%10);
                    tail = tail.next;
                }
                overflow = value>=10;
                l1 = l1==null?null:l1.next;
                l2 = l2==null?null:l2.next;
            }
            return(head);
        }
    }

File: 2. 两数相加/001.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
#include <stdlib.h>
struct ListNode {
    int val;
    struct ListNode *next;
};

typedef struct ListNode Node;

struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2){
    int value;
    int overflow = 0;
    Node* p1 = l1;
    Node* p2 = l2;
    Node* head = NULL;
    Node* tail = NULL;
    while( p1!=NULL && p2!=NULL ){
        value = p1->val + p2->val + overflow;
        overflow = value >= 10;
        if(head==NULL){
            head = (Node*)malloc(sizeof(Node));
            tail = head;
        }
        else{
            tail->next = (Node*)malloc(sizeof(Node));
            tail = tail->next;
        }
        tail->val = value % 10;
        p1 = p1->next;
        p2 = p2->next;
    }
    while( p1!=NULL ){
        value = p1->val + overflow;
        overflow = value >= 10;
        tail->next = (Node*)malloc(sizeof(Node));
        tail = tail->next;
        tail->val = value % 10;
        p1 = p1->next;
    }
    while( p2!=NULL ){
        value = p2->val + overflow;
        overflow = value >= 10;
        tail->next = (Node*)malloc(sizeof(Node));
        tail = tail->next;
        tail->val = value % 10;
        p2 = p2->next;
    }
    if(overflow){
        tail->next = (Node*)malloc(sizeof(Node));
        tail = tail->next;
        tail->val = overflow;
        tail->next = NULL;
    }
    else{
        tail->next = NULL;
    }
    return(head);
}